### ISSUE TITLE
Template 201-premium-storage-windows-vm broken because of boot diagnostics 

### ISSUE THREAD

**Issue by chevalpartners** [Azure/azure-quickstart-templates#673]
This template creates just 1 premium storage account of type Premium_LRS.
It uses this storage account to store the OS disk.
It also uses this account to set the boot diagnostics.

For whatever reason boot diagnostics does not support storage account of type Premium_LRS.
I got this error.

statusCode:Conflict
statusMessage:{"status":"Failed","error":{"code":"ResourceDeploymentFailure","message":"The resource operation completed with terminal provisioning state 'Failed'.","details":[{"code":"StorageAccountTypeNotSupported","message":"Storage account testupgradegs is of Premium_LRS type. Boot Diagnostics supports Standard_GRS,Standard_LRS,Standard_RAGRS storage account types."}]}}

I manually updated the template to use a storage account which was not of type Premium_LRS and my problem went away.

**Comment by kenazk** [comment]
Thanks for reporting. We decided to not enable boot diagnostics for Premium_LRS as each blob (between 0 and 127GB) is charged based on the tier (P1, P2, P3). Rather than let customers find out they have been paying for 3 blobs (OS VHD, screenshot, and serial output for Linux) vs 1, we decided to disallow Premium_LRS as an option for boot-diagnostics. As far as the template goes, I can update the templates so a standard storage account is always created, so you won't run into this issue unexpectedly.
